client/app/user/UserProfile.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
+import User from './userContainer';
+import userReducer from './userReducer';
+
+
+export default (props, _railsContext) => {
+  const store = createStore(userReducer);
```

Comment:
change to `createStore(userReducer, user: props.user);` and get rid of the if statement on lines 10-15